client/src/store/features/users/usersSlice.js

```diff
@@ -205,4 +205,24 @@ export function updateUserProfile(userId, userToken) {
   }
 }
 
+export const signupUser = createAsyncThunk(
```

Comment:
For the `signUpUser`, you can just create a simple thunk action rather than use the `createAsyncThunk`. 
```javascript
export const signUpUser = (username, email, password) => {
// return the async action that will call the dispatch function
   return async (dispatch) => {
     try {
// make the fetch request using the appropriate userService function
   const response = await userService.signUp(username, email, password);
   return response.data
} catch (error) {
// handle error
}
}
}
```

Reply:
Thanks, I'll be starting now from the simpler form. Later we can always upgrade if necessary.